### PR TITLE
chore: upgrade to webpack 5

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

utdated tooling packages due to webpack 4

security issue:
webpack4 => watchpack => ... => glob-parent

**What is the chosen solution to this problem?**

* Upgrade to webpack 5

QA validations tasks:
* yarn list --pattern webpack is as flat as possible ✅ 
* yarn list --pattern loader is as flat as possible ✅
* yarn postinstall ✅ (icons, tokens, theme are OK)
* yarn run pre-release ✅ (check dependencies.json are generated along report and map files)

storybook still use webpack 4. WIP on it to analyze the impact to use builder wepback5 on it.

**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,13 @@
...
   "resolutions": {
     "@types/react": "^16.14.25",
-    "@types/react-dom": "^16.9.14"
+    "@types/react-dom": "^16.9.14",
+    "webpack": "5.72.0",
+    "@storybook/react/webpack": "5.72.0",
+    "@storybook/core-common/webpack": "5.72.0",
+    "@storybook/core-server/webpack": "5.72.0",
+    "@storybook/builder-webpack4/webpack": "5.72.0",
+    "@storybook/manager-webpack4/webpack": "5.72.0"
```
Comment (Kporal) [inline package.json:43]: Missing caret

File: packages/design-system/talend-scripts.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
   "preset": "@talend/scripts-preset-react-lib",
   "copy": [{ "from": "src/images", "to": "images" }],
   "webpack": {
+    "debug": true,
```
Comment (Kporal) [inline packages/design-system/talend-scripts.json:5]: To be removed

File: tools/scripts-config-react-webpack/package.json [hunk 3]
```diff
@@ -22,33 +22,34 @@
...
+    "source-map-loader": "^3.0.0",
+    "style-loader": "^3.3.1",
     "svg64": "^1.1.0",
     "terser": "^5.12.1",
-    "terser-webpack-plugin": "^4.2.3",
+    "terser-webpack-plugin": "^5.3.0",
     "tmp": "^0.2.1",
-    "url-loader": "^3.0.0",
-    "webpack": "^4.46.0",
+    "url": "^0.11.0",
```
Comment (Kporal) [inline tools/scripts-config-react-webpack/package.json:50]: ⚠️ Can we do something about it?

**Comment by jmfrancois** [comment]
moved to #4005

**Pull request closed without merging**